daemon/endpoint.go

```diff
@@ -76,7 +78,9 @@ func (h *getEndpoint) Handle(params GetEndpointParams) middleware.Responder {
 		go func(wg *sync.WaitGroup, ep *endpoint.Endpoint) {
 			ep.Mutex.RLock()
 			if ep.HasLabels(convertedLabels) {
+				epListLock.Lock()
```

Comment:
I have a private branch that I've been working to hide the endpointmananger's mutex. On that branch this is re-written because of this bug. Can you please add a `FIXME @aanm`? So once I rebase, I'll remember what is the conflict about?

Reply:
I do not understand what you mean.`epListLock` is not related to `endpointmanager`; it protects the list of endpoints that is written to concurrently by the spawned goroutines. What do you mean by it is "re-written because of this bug"? I have not changed the use of the endpointmanager mutex.

Reply:
I have this changes:
```
	epChan := make(chan *models.Endpoint, len(eps))
	finishAppend := make(chan struct{})

	wg.Add(len(eps))
	for k := range eps {
		go func(wg *sync.WaitGroup, epChan chan<- *models.Endpoint, ep *endpoint.Endpoint) {
			if ep.HasLabels(convertedLabels) {
				epChan <- ep.GetModel()
			}
			wg.Done()
		}(&wg, epChan, eps[k])
	}

	go func() {
		for ep := range epChan {
			resEPs = append(resEPs, ep)
		}
		close(finishAppend)
	}()
	wg.Wait()
	close(epChan)
	<-finishAppend

	if params.Labels != nil && len(resEPs) == 0 {
		return NewGetEndpointNotFound()
	}

	return NewGetEndpointOK().WithPayload(resEPs)
```

Reply:
I'll just implement this here then in that case.

Reply:
Update: I've implemented @rlenglet 's functionality approach below as it results in me only needing one channel. I'll add the FIXME into my next push as it does not remove the `endpointmanager` mutex.